### DISCUSSION TITLE
Recover lost Android-based (direct / none) traffic

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#1337] [Anything Else]
![ezgif com-gif-maker-13](https://user-images.githubusercontent.com/34340819/133574910-0409ed64-4f91-4811-910f-00da900336b9.jpg)

I wonder if [a tweak like this](https://onlinemarkethink.com/recover-direct-traffic-google-analytics-full-referrer/) would be possible for us to do for all the traffic we track on Plausible

**Comment by ukutaht** [comment]
Related issue in the upstream project that we use for referrer parsing: https://github.com/snowplow-referer-parser/referer-parser/issues/131

Interestingly there doesn't seem tons of motivation to fix and one person suspects it's referrer spam. 

To me these `android-app` referrers seem totally legit and I think we should be parsing these as explained in the blog post. Currently we also classify these as Direct / None.

We don't need upstream to implement it, we could add our own rules on top. It's not very difficult to add this.

Great find @metmarkosaric.

**Reply by metmarkosaric** [reply]
nice! that would be a great feature! this trick is being spread around the marketing world and people are manually fixing it for themselves but i don't know any analytics tool that uncovers it by default. would be really useful!

**Comment by metmarkosaric** [comment]
"we switched to Plausible a few months ago. However, we're still using our own analytics because we're not able to filter Google Discover traffic. the referrer can be used to correctly identify Discover traffic. From a privacy perspective absolutely uncritical, from a publisher perspective a fantastic enhancement."

(feedback via email) 

It's also possible to track Google Discover traffic with this method. More at https://valentin.app/discover-tracking.html

"After applying the filter described later on you will see something similar to the following screenshot. Users from Google Discover will show up as www.google.com and com.google.android.googlequicksearchbox/ while traffic from www.google.com/ is Google Search."

**Comment by metmarkosaric** [comment]
**Update**: Starting from yesterday evening, we're attempting to uncover traffic originating from Android apps which is traditionally categorized as direct in Google Analytics and other analytics tools. The volume of "**android-app**" entries in your "Top Sources" report will vary based on your site and audience. For sites with significant mobile traffic, this could recover 10% or more previously unattributed traffic from apps like Gmail, Slack and Telegram.

**Comment by Filippo-Marcel-Lisuzzo** [comment]
hello, are there any updates regarding google discover traffic tracking?

**Comment by jeans** [comment]
Discover Referrer Tracking would be really awesome. Where did your attempts lead to @metmarkosaric ?

**Comment by metmarkosaric** [comment]
hey! we've looked into Google Discover but couldn't find a reliable way to differentiate Google Discover traffic from Google Search traffic so currently we include them both in the Google source. we will revisit it again if Google changes things. there's a thread for this on our feedback board: https://feedback.plausible.io/306
